tgui/packages/tgui/interfaces/PreferencesMenu/LoadoutManager.tsx

```diff
@@ -0,0 +1,406 @@
+    return null;
+  }
+
+  return (
+    <Button
+      icon={button.icon}
+      height="22px"
+      width="22px"
+      tooltip={button.tooltip}
+      tooltipPosition={'bottom-start'}
+      disabled={!button.act_key}
+      color="yellow"
+      style={{ zIndex: '2' }}
+      onClick={(e) => {
+        e.stopPropagation();
+        act('pass_to_loadout_item', {
+          path: item.path,
+          subaction: button.act_key,
+        });
+      }}
+    />
+  );
+};
+
+const ItemDisplayButtons = (props: { item: LoadoutItem; active: boolean }) => {
+  const { item, active } = props;
+  const buttons = item.buttons;
+  return (
+    <Flex>
+      {buttons.map((button) => (
+        <Flex.Item key={button.act_key} mr={1}>
+          <ItemDisplayButton item={item} button={button} active={active} />
+        </Flex.Item>
+      ))}
+    </Flex>
+  );
+};
+
+const ItemDisplay = (props: { item: LoadoutItem; active: boolean }) => {
+  const { act } = useBackend<LoadoutItem>();
+  const { item, active } = props;
+  return (
+    <Button
+      height="100px"
+      width="100px"
+      color={active ? 'green' : 'default'}
+      style={{ textTransform: 'capitalize', zIndex: '1' }}
+      tooltip={item.name}
+      tooltipPosition={'bottom'}
+      onClick={() =>
+        act('select_item', {
+          path: item.path,
+          deselect: active,
+        })
+      }
+    >
+      <Stack vertical>
+        <Stack.Item ml={-1}>
+          <ItemDisplayButtons item={item} active={active} />
+        </Stack.Item>
+        <Stack.Item mt={ShowsButtons(item, active) ? -4 : 0}>
+          <ItemIcon item={item} />
+        </Stack.Item>
+      </Stack>
+    </Button>
+  );
+};
+
+const ItemListDisplay = (props: { items: LoadoutItem[] }) => {
+  const { data } = useBackend<LoadoutManagerData>();
+  const loadout = data.character_preferences.misc.loadout_list;
+  return (
+    <Flex wrap>
+      {props.items.map((item) => (
+        <Flex.Item key={item.name} mr={2} mb={2}>
+          <ItemDisplay
+            item={item}
+            active={loadout && loadout[item.path] !== undefined}
+          />
+        </Flex.Item>
+      ))}
+    </Flex>
+  );
+};
+
+const LoadoutTabDisplay = (props: {
+  category: LoadoutCategory | undefined;
+}) => {
+  const { category } = props;
+  if (!category) {
+    return (
+      <NoticeBox>
+        Erroneous category detected! This is a bug, please report it.
+      </NoticeBox>
+    );
+  }
+
+  return <ItemListDisplay items={category.contents} />;
+};
+
+const SearchDisplay = (props: {
+  loadout_tabs: LoadoutCategory[];
+  currentSearch: string;
+}) => {
+  const { loadout_tabs, currentSearch } = props;
+
+  const allLoadoutItems = () => {
+    const concatItems: LoadoutItem[] = [];
+    for (const tab of loadout_tabs) {
+      for (const item of tab.contents) {
+        concatItems.push(item);
+      }
+    }
+    return concatItems.sort((a, b) => a.name.localeCompare(b.name));
```

Comment:
@stylemistake had some negative opinions on this, consider an a>b sort rather than localecompare